test/uk/gov/hmrc/merchandiseinbaggage/repositories/CryptoDeclarationRepositoryImplSpec.scala

```diff
@@ -0,0 +1,234 @@
+    }
+
+    "Update pre-encrypted declaration, simulate adding Amendment" in {
+      val declaration = aDeclaration
+      val repository = new DeclarationRepositoryImpl(reactiveMongo.mongoConnector.db)
+
+      // Write a non-encrypted declaration
+      repository.insertDeclaration(declaration).futureValue.encrypted mustBe None
+
+      // Find declaration (ie start of amendment)
+      val declarationAfterFindBy = whenReady(cryptoRepository.findBy(declaration.mibReference)) { findResult =>
+        findResult mustBe Some(declaration)
+        findResult.map(_.encrypted mustBe None)
+        findResult.getOrElse(fail)
+      }
+
+      // Add Amendment and update declaration
+      val declarationAfterFindByWithAmendment = declarationAfterFindBy.copy(amendments = Seq(aAmendment))
+      whenReady(cryptoRepository.upsertDeclaration(declarationAfterFindByWithAmendment)) { findResult =>
+        findResult.encrypted mustBe None
+      }
+
+      // Find and check it is now encrypted
+      whenReady(cryptoRepository.findBy(declarationAfterFindBy.mibReference)) { findResult =>
+        findResult.map { declarationWithEncryption =>
+          declarationWithEncryption.encrypted mustBe Some(true)
+          declarationWithEncryption.nameOfPersonCarryingTheGoods.firstName mustBe declaration.nameOfPersonCarryingTheGoods.firstName
+          declarationWithEncryption.nameOfPersonCarryingTheGoods.lastName mustBe declaration.nameOfPersonCarryingTheGoods.lastName
+          declarationWithEncryption.email mustBe declaration.email
+          declarationWithEncryption.eori mustBe declaration.eori
+        }
+      }
+    }
+  }
+
+  "crypto " should {
+    val declarationCrypto = injector.instanceOf[CryptoDeclarationRepositoryImpl]
+    val declaration = aDeclaration
+
+    "encrypt -> decrypt must match the original with encrypted enabled" in {
+      declarationCrypto.decryptDeclaration(declarationCrypto.encryptDeclaration(declaration)) mustBe declaration.copy(
+        encrypted = Some(true))
+    }
+
+    "encrypt name should be different from original" in {
+      declarationCrypto.encryptDeclaration(declaration).nameOfPersonCarryingTheGoods should not be declaration.nameOfPersonCarryingTheGoods
+    }
+
+    "encrypt email should be different from original" in {
+      declarationCrypto.encryptDeclaration(declaration).email should not be declaration.email
+    }
+
+    "encrypt eori should be different from original" in {
+      declarationCrypto.encryptDeclaration(declaration).eori should not be declaration.eori
+    }
+
+    "encrypt maybeCustomsAgent should be different from original" in {
+      val declarationWithCustomsAgent = declaration.copy(maybeCustomsAgent = Some(aCustomsAgent))
+
+      declarationCrypto
+        .encryptDeclaration(declarationWithCustomsAgent)
+        .maybeCustomsAgent should not be declarationWithCustomsAgent.maybeCustomsAgent
+    }
+
+    "encrypt journeyDetails should be different from original" in {
```

Comment:
in all the above tests, instead of checking the texts are different, i think the correct way to verify the encryption is by verifying the encrypted text, for ex: `encrypt(email) must Be "encryptedEmail"`. 
This way the tests would fail if someone changes the key or crypto algo

Reply:
done